lib/ui/instrumentpanel.js

```diff
@@ -261,7 +262,14 @@ InstrumentPanel.prototype.onWidgetChange = function(widget) {
 
 InstrumentPanel.prototype.persist = function() {
   if (this.currentPage !== notificationPageId) {
-    browser.saveGrid(this.host, this.serializableSettings());
+    var that = this;
```

Comment:
Just thinking here: would it be safer to call `serializableSettings()` when `persist` is invoked and not later? Is there danger that something has changed during the delay and is not meant to be saved yet?

Reply:
By design, all modifications on the layout (layout & widget options) need to be saved.
There is one case, `onLayoutChange` where layout is modified after widget options have changed.
But `persist` is called in last in `onLayoutChange` function and serialised once again and saved one second later.
Even if widget options differ from the layout (by ex : widget present and absent from the layout), a default layout for the widget will be generated by `getInitialLayout`.
And calling `serializableSettings` on each `persist` call have an extra cost of CPU usage.